### PR TITLE
UI - Resolve Github code scanning alerts

### LINKED ISSUE #1305: UI: Fix code scanning alerts
Tracking issue for:
- [x] https://github.com/vmware-tanzu/tanzu-framework/security/code-scanning/12
- [x] https://github.com/vmware-tanzu/tanzu-framework/security/code-scanning/13
- [x] https://github.com/vmware-tanzu/tanzu-framework/security/code-scanning/11
- [x] https://github.com/vmware-tanzu/tanzu-framework/security/code-scanning/3
- [x] https://github.com/vmware-tanzu/tanzu-framework/security/code-scanning/4

### PR DESCRIPTION
**Pull request by miclettej**
This PR resolves GH code scanning alerts raised on the UI code base

These alerts do not affect production code, but are being resolved
to keep the entire UI code base conformant with these scans.

Resolves:
Clear-text logging of sensitive information in E2E tests
Missing rate limit in Node.js mock API server

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by swalner-vmware — APPROVED** [review]
LGTM

**Review by saji-pivotal — APPROVED** [review]

**Comment by navidshaikh** [comment]
I think this is ready to go? @miclettej

**Comment by miclettej** [comment]
> I think this is ready to go? @miclettej
It needs approval by a code owner. I'll ping the group again now.

**Pull request merged**
